File: leetcode-04/solutions/python3/0565.py
class Solution:
    def arrayNesting(self, nums: List[int]) -> int:
        ans = 0

        for num in nums:
            if num != float('inf'):
                start = num
                count = 0
                while nums[start] != float('inf'):
                    temp = start
                    start = nums[start]
                    nums[temp] = float('inf')
                    count += 1
                ans = max(ans, count)

        return ans

File: leetcode-04/solutions/cpp/0565.cpp
class Solution {
 public:
  int arrayNesting(vector<int>& nums) {
    int ans = 0;

    for (int num : nums)
      if (num != INT_MAX) {
        int start = num;
        int count = 0;
        while (nums[start] != INT_MAX) {
          int temp = start;
          start = nums[start];
          nums[temp] = INT_MAX;
          count++;
        }
        ans = max(ans, count);
      }

    return ans;
  }
};